ml-agents/mlagents/trainers/trainer.py

```diff
@@ -248,6 +245,13 @@ def advance(self) -> None:
         """
         Steps the trainer, taking in trajectories and updates if ready.
         """
+        if (
```

Comment:
I think you'd have less "drift" in the reporting steps if you did something like
``` python
def __init__(...):
  self.summary_freq = self.trainer_parameters["summary_freq"]
  self.next_update_step = self.summary_freq

def end_episode(self):
  if self.get_step > self.next_update_step:
    self._write_summary()
    # next update will be the next multiple of summary_freq
    self.next_update_step = self.get_step + (summary_freq - self.get_step % summary_freq)
```

Reply:
Moved logic to process_trajectory